feature-libs/product/image-zoom/components/product-image-zoom/product-image-zoom-product-images/product-image-zoom-product-images.component.html

```diff
@@ -1,5 +1,11 @@
 <ng-container *ngIf="mainImage$ | async as main">
-  <cx-media [container]="main" (click)="triggerZoom(true)"></cx-media>
+  <ng-container *cxLcpContext="let lcpContext">
+    <cx-media
+      [container]="main"
+      (click)="triggerZoom(true)"
+      [fetchPriority]="lcpContext.fetchPriority$ | async"
+    ></cx-media>
+  </ng-container>
```

Comment:
`query:` Have you perhaps considered using the structural directive directly on the target component?
```suggestion
    <cx-media
      *cxLcpContext="let lcpContext"
      [container]="main"
      (click)="triggerZoom(true)"
      [fetchPriority]="lcpContext.fetchPriority$ | async"
    ></cx-media>
```

Reply:
Yes I considered it. And decided to do it in ng-container for readability - to not mix it with other inputs on the `<cx-media>` element.

Reply:
My take - using the structural directive directly on the target component looks to me more natural, and it's something we used to when using Angular's structural directives. (In addition, it provides less boilerplate)